tests/test_ingesters.py

```diff
@@ -695,6 +694,58 @@ def test_parameter_assignment_for_attributes(self):
         self.assertEqual(created_dataset_uri, False)
 
 
+class FTPIngesterTestCase(django.test.TestCase):
+    """Test the FTPIngester"""
+
+    def setUp(self):
+        self.patcher_param_count = mock.patch.object(Parameter.objects, 'count')
+        self.mock_param_count = self.patcher_param_count.start()
+        self.mock_param_count.return_value = 2
+
+    def test_function_get_normalized_attributes_ceda(self):
+        """ test the functionality of '_get_normalized_attributes' for a FTPIngester.
+        Keys must be equal to DATASET_CUMULATIVE_PARAMETER_NAMES plus DATASET_PARAMETER_NAMES plus
+        two additional new keys.
+        Also a "None" value should not be in the resulted normalized attributes. """
+        input_url = 'ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/data/CDR_v2/Climatology/L4/v2.1/D365-ESACCI-L4_GHRSST-SSTdepth-OSTIA-GLOB_CDR2.1-v02.0-fv01.0.nc'
+
+        ingester = ingesters.FTPIngester()
+        normalized_attributes = ingester._get_normalized_attributes(input_url)
+        self.assertCountEqual(list(normalized_attributes.keys()),
+                              ingester.DATASET_CUMULATIVE_PARAMETER_NAMES +
+                              ingester.DATASET_PARAMETER_NAMES +
+                              ['geospaas_service_name']+['geospaas_service']+['entry_id'])
```

Comment:
Please check that keys of `normalized_attributes` are equal to a predefined list rather than just count. And in the methods below

Reply:
@akorosov 
that `assertCountEqual` checks that issue:
![image](https://user-images.githubusercontent.com/62345618/92592990-ac710a80-f2a0-11ea-8fa3-6a9e732bdb04.png)

Only the name of it is misleading! Order does not matter for it which the main reason why I use it in the test.

above snapshot is from: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

Reply:
OK, great! I'll reuse it :)